### PR TITLE
Adapt kernelvethpair to sendfd recvfd

### PR DESCRIPTION
**Pull request by edwarnicke**
Note: Still to do: kerneltap, memif, directmemif

### CODE REVIEW

**Comment by edwarnicke** [comment]
Can't be properly updated to work until https://github.com/networkservicemesh/sdk/pull/467 is merged

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go [hunk 1]
```diff
@@ -21,37 +21,33 @@ package kernelvethpair
...
 // NewClient provides NetworkServiceClient chain elements that support the kernel Mechanism using veth pairs
 func NewClient() networkservice.NetworkServiceClient {
-	return &kernelVethPairClient{
-		fileNameFromInodeNumberFunc: netnsinode.LinuxNetNSFileName,
-	}
+	return &kernelVethPairClient{}
 }
 
 // NewTestableClient - same as NewClient, but allows provision of fileNameFromInodeNumberFunc to allow for testing
 func NewTestableClient(fileNameFromInodeNumberFunc func(string) (string, error)) networkservice.NetworkServiceClient {
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go:49]: fileNameFromInodeNumberFunc is unused, consider removing it. Do we need NewTestableClient at all?

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go [hunk 2]
```diff
@@ -60,23 +56,39 @@ func (k *kernelVethPairClient) Request(ctx context.Context, request *networkserv
 		Type: kernel.MECHANISM,
 	}
 	request.MechanismPreferences = append(request.MechanismPreferences, preferredMechanism)
-	conn, err := next.Client(ctx).Request(ctx, request, opts...)
+	var p peer.Peer
+	conn, err := next.Client(ctx).Request(ctx, request, append(opts, grpc.Peer(&p))...)
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go:60]: Why we need a peer.Peer here? I do not see it is used.

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go [hunk 3]
```diff
@@ -60,23 +56,39 @@ func (k *kernelVethPairClient) Request(ctx context.Context, request *networkserv
...
+	if err := k.appendInterfaceConfig(ctx, conn); err != nil {
 		return nil, err
 	}
 	return conn, nil
 }
 
 func (k *kernelVethPairClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
-	rv, err := next.Client(ctx).Close(ctx, conn, opts...)
+	var p peer.Peer
+	rv, err := next.Client(ctx).Close(ctx, conn, append(opts, grpc.Peer(&p))...)
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go:72]: Same here, do not see Peer is used.

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/server.go [hunk 4]
```diff
@@ -22,48 +22,61 @@ package kernelvethpair
...
+type kernelVethPairServer struct{}
 
 // NewServer provides NetworkServiceServer chain elements that support the kernel Mechanism using veth pairs
 func NewServer() networkservice.NetworkServiceServer {
-	return &kernelVethPairServer{fileNameFromInodeNumberFunc: netnsinode.LinuxNetNSFileName}
+	return &kernelVethPairServer{}
 }
 
 // NewTestableServer - same as NewServer, but allows provision of fileNameFromInodeNumberFunc to allow for testing
 func NewTestableServer(fileNameFromInodeNumberFunc func(string) (string, error)) networkservice.NetworkServiceServer {
```
Comment (haiodo) [inline pkg/networkservice/mechanisms/kernel/kernelvethpair/server.go:47]: fileNameFromInodeNumberFunc is unused, do we really need NewTestableServer?

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go (hunk 1 above)
Comment (edwarnicke) [reply pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go]: Good catch.  You are correct, we don't need NewTestableClient at all anymore :)  Fixed

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go (hunk 2 above)
Comment (edwarnicke) [reply pkg/networkservice/mechanisms/kernel/kernelvethpair/client.go]: You are correct, that's an artifact of a previous approach that should have been trimmed from the code.  Removed.

File: pkg/networkservice/mechanisms/kernel/kernelvethpair/server.go (hunk 4 above)
Comment (edwarnicke) [reply pkg/networkservice/mechanisms/kernel/kernelvethpair/server.go]: You are correct, that's an artifact of a previous approach that should have been trimmed from the code. Removed.

**Review by haiodo — APPROVED** [review]

**Pull request merged**